src/snowflake/cli/api/project/schemas/entities/entities.py

```diff
@@ -56,9 +56,5 @@
 
 v2_entity_model_types_map = {e.get_type(): e for e in ALL_ENTITY_MODELS}
 v2_entity_model_to_entity_map: Dict[EntityModel, Entity] = {
-    ApplicationEntityModel: ApplicationEntity,
-    ApplicationPackageEntityModel: ApplicationPackageEntity,
-    StreamlitEntityModel: StreamlitEntity,
-    FunctionEntityModel: FunctionEntity,
-    ProcedureEntityModel: ProcedureEntity,
+    e.get_entity_model_type(): e for e in ALL_ENTITIES
```

Comment:
I'm finding it difficult to understand why we have to resort to union types to define the entity and model concepts. One of these days I'd really like it if you could go over it with me.